crashreports/crashreporter/CrashReportApp.cpp

```diff
@@ -0,0 +1,464 @@
+            commentCtrl->SetMaxLength(MaxUserCommentLength);
+        }
+
+        if (onSend != nullptr)
+        {
+            auto okButton = new wxButton(frame, wxID_ANY, XC("&Don't send", "crash reporter button"));
+            auto sendButton = new wxButton(frame, wxID_ANY, XC("&Send", "crash reporter button"));
+
+            okButton->Bind(wxEVT_BUTTON, [frame](wxCommandEvent&)
+                {
+                    frame->Close(true);
+                });
+            sendButton->Bind(wxEVT_BUTTON, [frame, commentCtrl, onSend](wxCommandEvent&)
+                { 
+                    if (onSend(commentCtrl->GetValue()))
+                    {
+                        frame->Close(true);
+                    }
+                });
+
+            buttonsLayout->Add(okButton);
+            buttonsLayout->AddSpacer(5);
+            buttonsLayout->Add(sendButton);
+        }
+        else
+        {
+            auto okButton = new wxButton(frame, wxID_OK, wxT("OK"));
+            okButton->Bind(wxEVT_BUTTON, [frame](wxCommandEvent&)
+                {
+                    frame->Close(true);
+                });
+            buttonsLayout->Add(okButton);
+        }
+
+        mainLayout->Add(headerLayout, wxSizerFlags().Border(wxALL));
+        if (onSend != nullptr)
+        {
+            mainLayout->AddSpacer(5);
+            mainLayout->Add(new wxStaticText(frame, wxID_ANY, _("Click \"Send\" to submit the report to Audacity. This information is collected anonymously.")), wxSizerFlags().Border(wxALL));
+        }
+        mainLayout->AddSpacer(10);
+        mainLayout->Add(new wxStaticText(frame, wxID_ANY, _("Problem details")), wxSizerFlags().Border(wxALL));
+        
+        auto dumpTextCtrl = new wxTextCtrl(frame, wxID_ANY, dump, wxDefaultPosition, wxSize(500, 300), wxTE_RICH | wxTE_READONLY | wxTE_MULTILINE | wxTE_DONTWRAP);
+        dumpTextCtrl->SetFont(wxFont(wxFontInfo().Family(wxFONTFAMILY_TELETYPE)));
+        dumpTextCtrl->ShowPosition(0);//scroll to top
+        mainLayout->Add(dumpTextCtrl, wxSizerFlags().Border(wxALL).Expand());
+
+        if (onSend != nullptr)
+        {
+            mainLayout->AddSpacer(10);
+            mainLayout->Add(new wxStaticText(frame, wxID_ANY, _("Comments")), wxSizerFlags().Border(wxALL));
+            mainLayout->Add(commentCtrl, wxSizerFlags().Border(wxALL).Expand());
+        }
+
+        mainLayout->Add(buttonsLayout, wxSizerFlags().Border(wxALL).Align(wxALIGN_RIGHT));
+        frame->SetSizerAndFit(mainLayout);
+
+        frame->Show(true);
+    }
+}
+
+
+bool CrashReportApp::OnInit()
+{
+    if (!wxApp::OnInit())
+        return false;
+    
+    if (mSilent)
+    {
+        if (!mURL.empty())
+            SendMinidump(mURL, mMinidumpPath, mArguments, wxEmptyString);
+    }
+    else
+    {
+        static std::unique_ptr<wxLocale> sLocale(new wxLocale(wxLANGUAGE_DEFAULT));
```

Comment:
Finally, this use of new is unlike the others.  It isn't "naked" because it is the constructor argument of unique_ptr.

Still I prefer not having the keyword new appear at all, and would write instead:

```
static auto sLocale = std::make_unique<wxLocale>(wxLANGUAGE_DEFAULT);
```

Which is also briefer to read, avoiding repetition of the type name `wxLocale`.

Understand that all these remarks are not just about style:  they are about the right habits for writing leak-free code using idioms consistent with much of Audacity.  Especially point 2 above.  It's proper "RAII".  It means I don't have to examine later lines of code to see whether they might throw exceptions or return early before the pointer is passed along to some other owner.

This completes my comments about allocations.